### DISCUSSION TITLE
Custom pages with NextAuth v5 and RSC?

### DISCUSSION THREAD

**Discussion by chrislgrigg** [nextauthjs/next-auth#9384] [Help]
Hello! Does anyone have any advice or examples of using NextAuth v5 beta 4 or later to create a custom signin page using Next.js13/14 RSC? The [example in the docs](https://authjs.dev/guides/basics/pages#email-sign-in) uses the Pages router and, more importantly, `next-auth/react` does not appear to export `getCsrfToken` and I can't find a modern equivalent.

The [big discussion about v5](https://github.com/nextauthjs/next-auth/discussions/8487) does have some mentions of getting a CSRF token by hacking it out of cookies but is imprecise and seemingly undocumented so I don't want to rely on it. Is there an official or verified trustworthy "right way" to do this?

**Comment by shlroland** [comment]
same issue
